### ISSUE TITLE
Switch aria-label is covered by another element

### ISSUE THREAD

**Issue by EllaSharakanski** [chakra-ui/chakra-ui#3955]
# 🐛 Bug report

When using the `Switch` component with an `aria-label` property, the aria-label is applied to the `input` element, but this element is covered by other elements. When using `cypress` for example to do `cy.get('[aria-label="my-switch-label"]').click()`, it doesn't work because it throws an exception saying that the `input` element is being covered by another element so it cannot be clicked.

## 💥 Steps to reproduce

1. Go to https://chakra-ui.com/docs/form/switch
2. Add an aria-label to one of the switch components, like so: `<Switch colorScheme="red" aria-label="bla" />` (colorScheme is optional).
3. Right-click on the switch component and open the dev-tools.
4.  You'll see something like:
```
<div class="css-1axwbpm">
  <label class="chakra-switch css-1glqkmk">
    <input class="chakra-switch__input" type="checkbox" aria-label="bla" value="" style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;">. 
    <span aria-hidden="true" class="chakra-switch__track css-yaew4a">
      <span class="chakra-switch__thumb css-1t2ag9e"></span>
    </span>
  </label>
</div>
```
Note how the `aria-label` is on the `input` element and not the `label` element.

## 🧐 Expected behavior

In a previous version of Chakra (I'm not sure which, I know that it worked well on Chakra 1.0.4), the `aria-label` was a property of the `label` element. You can see it in this codesandbox: https://codesandbox.io/s/homepage-forked-xf4qg?file=/src/App.tsx

When viewing the switch in dev-tools, it shows:
```
<div id="root">
  <label aria-label="blabla" class="chakra-switch css-r74jfy">
    <input class="chakra-switch__input" type="checkbox" value="" style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;">
    <span aria-hidden="true" class="chakra-switch__track css-5femyx" data-active="">
      <span class="chakra-switch__thumb css-uh46nq"></span>
    </span>
  </label>
</div>
```

| Software         | Version(s) |
| ---------------- | ---------- |
| Chakra UI        |       1.6.0     |
| Browser          |       Chrome 90     |
| Operating System |      Mac      |

**Comment by segunadebayo** [comment]
Hi @EllaSharakanski,

This was done to improve the accessibility of the switch, `aria-label` belongs to the `input`, not the label. See the backstory: #3480.

Feel free to use `data-qa` or `data-testid` as the attribute to select the `label` and use it in cypress.

**Comment by EllaSharakanski** [comment]
@segunadebayo Thanks. If I'm wrapping the `Switch` component with a `FormControl` component and I set the `aria-label` on the `FormControl` then the label appears on the `input` again. Is this a bug then?
